Source/displayinfo/include/ExtendedDisplayIdentification.h

```diff
@@ -124,22 +187,327 @@ namespace Plugin {
+                                colorSpaceMap |= static_cast<uint16_t>(ColorSpaceType::XVYCC_601);
+                            }
+                            if(dataBlock.Current()[2]  & (1 << 1)) {
+                                colorSpaceMap |= static_cast<uint16_t>(ColorSpaceType::XVYCC_709);
+                            }
+                            if(dataBlock.Current()[2]  & (1 << 2)) {
+                                colorSpaceMap |= static_cast<uint16_t>(ColorSpaceType::SYCC_601);
+                            }
+                            if(dataBlock.Current()[2]  & (1 << 3)) {
+                                colorSpaceMap |= static_cast<uint16_t>(ColorSpaceType::OP_YCC_601);
+                            }
+                            if(dataBlock.Current()[2]  & (1 << 4)) {
+                                colorSpaceMap |= static_cast<uint16_t>(ColorSpaceType::OP_RGB);
+                            }
+                            if(dataBlock.Current()[2]  & (1 << 5)) {
+                                colorSpaceMap |= static_cast<uint16_t>(ColorSpaceType::ITUR_BT_2020_CYCC);
+                            }
+                            if(dataBlock.Current()[2]  & (1 << 6)) {
+                                colorSpaceMap |= static_cast<uint16_t>(ColorSpaceType::ITUR_BT_2020_YCC);
+                            }
+                            if(dataBlock.Current()[2]  & (1 << 7)) {
+                                colorSpaceMap |= static_cast<uint16_t>(ColorSpaceType::ITUR_BT_2020_RGB);
+                            }
+                            if(dataBlock.Current()[3]  & (1 << 7)) {
+                                colorSpaceMap |= static_cast<uint16_t>(ColorSpaceType::DCI_P3);
+                            }
+                        }
+                    }
+                }
+            }
+
+            void SupportedTimings(std::vector<uint8_t>& vicList)
+            {
+                DataBlockIterator dataBlock = DataBlockIterator(_segment, DetailedTimingDescriptorStart());
+                while(dataBlock.Next()) {
+                    if(dataBlock.IsValid() && (dataBlock.BlockTag() == 0x02)) {
+                        for(uint8_t index = 0; index < dataBlock.BlockSize(); index++) {
+                            const uint8_t vic = dataBlock.Current()[index] & 0x7F;
+                            vicList.push_back(vic);
+                        }
+                    }
+                }
+            }
+
+            void SupportedAudioFormats(uint32_t& audioFormatMap)
+            {
+                DataBlockIterator dataBlock = DataBlockIterator(_segment, DetailedTimingDescriptorStart());
+                while(dataBlock.Next()) {
+                    if(dataBlock.IsValid() && (dataBlock.BlockTag() == 0x01)) {
+                        uint8_t sadCount = dataBlock.BlockSize() % 3;
+                        for(uint8_t index = 0; index < dataBlock.BlockSize(); index+=3) {
```

Comment:
There's a possibility you jump beyond the buffer here

Reply:
data.BlockSize() will return 0 if the index is nor in range thus index should never index out